rmc/utils/generic.py

```diff
@@ -338,19 +339,35 @@ def get_ref_aa(
         )
 
     # Double check that protein start always equals protein end
-    protein_num_check = ht.aggregate(
-        hl.agg.count_where(ht.aa_start_num != ht.aa_end_num)
+    protein_num_check_he_path = f"{TEMP_PATH_WITH_FAST_DEL}/protein_num_count.he"
+    if overwrite_temp:
+        # Will always overwrite protein num check if overwrite_temp is set
+        # even if file already exists
+        protein_num_check = ht.aggregate(
+            hl.agg.count_where(ht.aa_start_num != ht.aa_end_num)
+        )
+        hl.experimental.write_expression(
+            protein_num_check, protein_num_check_he_path, overwrite=True
+        )
+
+    protein_num_check = hl.eval(
+        hl.experimental.read_expression(protein_num_check_he_path)
```

Comment:
hmm I don't see why this is better? this suggestion would have to be:
```
    if not overwrite_temp and file_exists(protein_num_check_he_path):
        protein_num_check = hl.eval(
            hl.experimental.read_expression(protein_num_check_he_path)
        )
    else:
        protein_num_check = ht.aggregate(
            hl.agg.count_where(ht.aa_start_num != ht.aa_end_num)
        )
        hl.experimental.write_expression(
            protein_num_check, protein_num_check_he_path, overwrite=overwrite_temp
        )
        protein_num_check = hl.eval(
            hl.experimental.read_expression(protein_num_check_he_path)
        )
```
to work, right?

Reply:
ah, is the suggestion is to avoid the unnecessary aggregation if this does already exist?

Reply:
will change to
```
    # Double check that protein start always equals protein end
    protein_num_check_he_path = f"{TEMP_PATH_WITH_FAST_DEL}/protein_num_count.he"
    if not file_exists(protein_num_check_he_path):
        protein_num_check = ht.aggregate(
            hl.agg.count_where(ht.aa_start_num != ht.aa_end_num)
        )
        hl.experimental.write_expression(
            protein_num_check, protein_num_check_he_path, overwrite=True
        )

    # Assume file already exists otherwise
    protein_num_check = hl.eval(
        hl.experimental.read_expression(protein_num_check_he_path)
    )
```

Reply:
Oh I thought we would write it regardless if `overwrite_temp` is True?

Reply:
Rather, I meant that if the file already exists but we want to overwrite it so `overwrite_temp` is True, your suggestion won't overwrite

Reply:
oh merp, you're right

Reply:
fixed!

Reply:
The new structure is fine with me and looks cleaner, but there are places where we have the old structure in the code e.g. for the LoFTEE agg